source/common/http/filter_manager.cc

```diff
@@ -1253,14 +1255,18 @@ void ActiveStreamDecoderFilter::setDecoderBufferLimit(uint32_t limit) {
 
 uint32_t ActiveStreamDecoderFilter::decoderBufferLimit() { return parent_.buffer_limit_; }
 
-bool ActiveStreamDecoderFilter::recreateStream() {
+bool ActiveStreamDecoderFilter::recreateStream(const ResponseHeaderMap* headers) {
   // Because the filter's and the HCM view of if the stream has a body and if
   // the stream is complete may differ, re-check bytesReceived() to make sure
   // there was no body from the HCM's point of view.
   if (!complete() || parent_.stream_info_.bytesReceived() != 0) {
     return false;
   }
 
+  if (headers != nullptr) {
+    parent_.filter_manager_callbacks_.chargeStats(*headers);
+  }
```

Comment:
You mention in the description "classified as 0xx", which stat are you referring to?

Are there some stats that track the number of internal redirects processed by cluster?

Reply:
The access_log sees 0 as response code when emit log is called.

Reply:
I see no tests that verify contents of the access log.

Reply:
Added checks for this in redirect_integration_test

Reply:
My most recent updates added more detailed checks of the access log.

I bypassed updating the downstream stats in chargeStats for internal redirect, and set the response headers so that they got logged in the access log as well. I updated the tests to reflect these changes.